### PR TITLE
.githooks: add hook to check go version conistency.

### PR DESCRIPTION
**Pull request by klihub**
Add a hook to check the consistency of declared golang version requirement vs. the one used in github workflows. No need to always wait for the already slow and overloaded github workers to run our tests till failure then let us know we forgot to update the other.

### CODE REVIEW

File: .githooks/pre-commit.d/20-go-version [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+WORKFLOWS=".github/workflows/verify.yml"
+
+gomod=$(git diff --cached go.mod | grep '^+go ' | sed 's/^.* //g')
```
Comment (marquiz) [inline .githooks/pre-commit.d/20-go-version:5]: Would it be good to catch the problem even if not part of this commit, i.e. just grep `go.mod` directly without doing git diff(?)

**Review by marquiz — DISMISSED** [review]
One question but looks good in this form, too

File: .githooks/pre-commit.d/20-go-version (hunk 1 above)
Comment (klihub) [reply .githooks/pre-commit.d/20-go-version:5]: Hmm... maybe. Should we then also always automagically enable our git hooks in the cloned repos (by making `install-git-hooks` a prerequisite of the `all` or `build` Makefile target). Otherwise the whole test is only run, and such mistakes are caught, by those (yours, truly) who enable the git hooks by running `make install-git-hooks`... Then again, if we enable the hooks for everyone, how those versions could go out of sync ever again, other people brutally bypassing all hooks using `git commit -n` ?

**Review by jukkar — DISMISSED** [review]

**Comment by klihub** [comment]
@marquiz @jukkar Rebased on latest master to CI passed, PTAL, and give a thumbs up/merge...

**Review by jukkar — APPROVED** [review]
